From ad8bf2ae08919f3fbf2715d7ee638176a8fdea54 Mon Sep 17 00:00:00 2001
From: kxxt <rsworktech@outlook.com>
Date: Sat, 10 Feb 2024 10:10:53 +0800
Subject: [PATCH] sysroot: riscv64 support

Patch-Dir: src/electron/patches/chromium
Patch-Filename: 0008-sysroot-riscv64-support.patch
---
 build/config/sysroot.gni                       | 4 +++-
 build/linux/sysroot_scripts/install-sysroot.py | 8 +++++---
 build/linux/sysroot_scripts/sysroots.json      | 7 +++++++
 3 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/build/config/sysroot.gni b/build/config/sysroot.gni
index 47fe4cbc840dd..965890580f417 100644
--- a/build/config/sysroot.gni
+++ b/build/config/sysroot.gni
@@ -22,7 +22,7 @@ declare_args() {
   use_sysroot =
       current_cpu == "x86" || current_cpu == "x64" || current_cpu == "arm" ||
       current_cpu == "arm64" || current_cpu == "mipsel" ||
-      current_cpu == "mips64el" || (current_cpu == "riscv64" && is_android)
+      current_cpu == "mips64el" || current_cpu == "riscv64"
 }
 
 if (sysroot == "") {
@@ -49,6 +49,8 @@ if (sysroot == "") {
       sysroot = "$target_sysroot_dir/debian_bullseye_armhf-sysroot"
     } else if (current_cpu == "arm64") {
       sysroot = "$target_sysroot_dir/debian_bullseye_arm64-sysroot"
+    } else if (current_cpu == "riscv64") {
+      sysroot = "$target_sysroot_dir/debian_sid_riscv64-sysroot"
     } else {
       assert(false, "No linux sysroot for cpu: $target_cpu")
     }
diff --git a/build/linux/sysroot_scripts/install-sysroot.py b/build/linux/sysroot_scripts/install-sysroot.py
index b1f6b17afd1af..081e140bf523d 100755
--- a/build/linux/sysroot_scripts/install-sysroot.py
+++ b/build/linux/sysroot_scripts/install-sysroot.py
@@ -37,7 +37,7 @@ SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
 SRC_DIR = os.path.dirname(os.path.dirname(os.path.dirname(SCRIPT_DIR)))
 
 VALID_ARCHS = ("amd64", "i386", "armhf", "arm64", "mipsel", "mips64el",
-               "ppc64el")
+               "ppc64el", "riscv64")
 
 ARCH_TRANSLATIONS = {
     "x64": "amd64",
@@ -91,14 +91,16 @@ def main(args):
         sysroots_json_path = DEFAULT_SYSROOTS_PATH
 
     if options.arch:
+        target_platform = 'sid' if options.arch == 'riscv64' else DEFAULT_TARGET_PLATFORM
         InstallSysroot(
             sysroots_json_path,
-            DEFAULT_TARGET_PLATFORM,
+            target_platform,
             ARCH_TRANSLATIONS.get(options.arch, options.arch),
         )
     elif options.all:
         for arch in VALID_ARCHS:
-            InstallSysroot(sysroots_json_path, DEFAULT_TARGET_PLATFORM, arch)
+            target_platform = 'sid' if arch == 'riscv64' else DEFAULT_TARGET_PLATFORM
+            InstallSysroot(sysroots_json_path, target_platform, arch)
     else:
         print("You much specify one of the options.")
         return 1
diff --git a/build/linux/sysroot_scripts/sysroots.json b/build/linux/sysroot_scripts/sysroots.json
index 2aade5a60861a..8ae67df7bfa53 100644
--- a/build/linux/sysroot_scripts/sysroots.json
+++ b/build/linux/sysroot_scripts/sysroots.json
@@ -40,5 +40,12 @@
         "SysrootDir": "debian_bullseye_ppc64el-sysroot",
         "Tarball": "debian_bullseye_ppc64el_sysroot.tar.xz",
         "URL": "https://commondatastorage.googleapis.com/chrome-linux-sysroot"
+    },
+    "sid_riscv64": {
+        "Key": "20240131T153825Z-4",
+        "Sha1Sum": "d8849d433e2c857b2a052812135115c91f96f54f",
+        "SysrootDir": "debian_sid_riscv64-sysroot",
+        "Tarball": "debian_sid_riscv64_sysroot.tar.xz",
+        "URL": "https://github.com/riscv-forks/electron/releases/download/sysroot-20240131T153825Z-4/debian_sid_riscv64_sysroot.tar.xz"
     }
 }
-- 
2.39.2

